### PR TITLE
Things, that didn't made to 2017.7 for some reasons

### PR DESCRIPTION
**Pull request by isbm**
### What does this PR do?

This is some "leftovers" that we carry on from version to version, but somehow they never get into the upstream (sometimes forgotten and yet-to-go, sometimes forgotten to upstream etc). Mostly these are `.patch` files [in our OBS package for current and next releases](https://build.opensuse.org/package/show/systemsmanagement:saltstack:products:next/salt).

It would be nice if we get rid of those and upstram just finally have them (as they anyway meant to be here).

### Tests written?

Yes. Sort of. :laughing:

### CODE REVIEW

**Comment by isbm** [comment]
@cachedout I'd ask you to help this get into the 2017.7, so we can get rid of those "leftovers". Sometimes we open PR to `YYYY.MM` version, but it never gets into the `develop`, although supposed to. Sometimes we open against `develop` and forgetting to back port it. So, can it happen we just merge this?

**Comment by isbm** [comment]
@cachedout This seems has some other little "leftovers" on lint, but I did not included these patches, since they will go with another PR. So if you say _"Good to go in 2017.7"_, then I would continue working on this one to let it pass through.

Otherwise just close it (and make me sad). :wink:

File: setup.py [hunk 1]
```diff
@@ -697,8 +697,7 @@ def run(self):
 
 
 INSTALL_VERSION_TEMPLATE = '''\
-# This file was auto-generated by salt's setup on \
-{date:%A, %d %B %Y @ %H:%m:%S UTC}.
```
Comment (garethgreenaway) [inline setup.py]: Particular reason to remove the time stamp of when the file is generated?

**Comment by garethgreenaway** [comment]
Looks like some lint issues and some others for the names of the zypper related tests.

File: setup.py (hunk 1 above)
Comment (isbm) [inline setup.py]: Well, the [corresponding commit](https://github.com/saltstack/salt/pull/41795/commits/0c6611f274b95d42055f62786c82a9812e8c0a1d) says all by itself. We had a [registered bug in our Bugzilla](https://bugzilla.suse.com/show_bug.cgi?id=969407) for this.

File: salt/modules/yumpkg.py [hunk 2]
```diff
@@ -924,6 +925,35 @@ def list_upgrades(refresh=True, **kwargs):
 list_updates = salt.utils.alias_function(list_upgrades, 'list_updates')
 
 
+def list_downloaded():
+    '''
+    .. versionadded:: Oxygen
```
Comment (mirceaulinic) [inline salt/modules/yumpkg.py]: This one seems to be marked to be added in Oxygen... guess we should rename?

File: salt/modules/yumpkg.py [hunk 3]
```diff
@@ -2915,3 +2962,64 @@ def diff(*paths):
...
+        patches[advisory_id] = {
+            'installed': True if inst == 'i' else False,
+            'summary': pkg
+        }
+    return patches
+
+
+def list_patches(refresh=False):
+    '''
+    .. versionadded:: Oxygen
```
Comment (mirceaulinic) [inline salt/modules/yumpkg.py]: Same silly comment here

File: salt/modules/zypper.py [hunk 4]
```diff
@@ -1864,6 +1878,35 @@ def download(*packages, **kwargs):
     )
 
 
+def list_downloaded():
+    '''
+    .. versionadded:: Oxygen
```
Comment (mirceaulinic) [inline salt/modules/zypper.py]: And here 😄

**Comment by isbm** [comment]
@meaksh are those "Oxygen" version added correct, or?

**Comment by meaksh** [comment]
@isbm - We added those "Oxygen" version because the initial PR was done against `develop` so it was targeted to be included on `Oxygen`, but if now we're including this into `Nitrogen` then I would say those version added tags need to be modified.

**Comment by cachedout** [comment]
Once we get these `Oxygen` tags fixed this should be ready.

**Comment by cachedout** [comment]
@isbm We really want to tag the RC today. Any chance you can make these changes in the next few hours?

**Comment by isbm** [comment]
@cachedout What Oxygen tags? I replaced them with Nitrogen and... _tags_?? I am confused.

**Comment by cachedout** [comment]
I was referring to this one:

https://github.com/saltstack/salt/pull/41795/files#diff-0939df492f4987da56df53de1335fdb6R1962

and this one:

https://github.com/saltstack/salt/pull/41795/files#diff-0939df492f4987da56df53de1335fdb6R1985

**Comment by isbm** [comment]
@cachedout uhm... refresh the browser? :laughing: 
https://github.com/saltstack/salt/pull/41795/commits/9a74c0a6f345517898a44074812220779a422f2f

**Comment by isbm** [comment]
@cachedout Is everything OK as it is right now? Because our today is already your tomorrow...

**Comment by cachedout** [comment]
@isbm That commit does not include either one of the lines I just linked you to. :]

Lines 1962 and 1985 in zypper.py.

**Comment by isbm** [comment]
@cachedout that must do the job.

**Comment by isbm** [comment]
@cachedout these Oxygen labels there still because all these patches are of that age... This is what happens, if you do not upstream stuff. Is anything else needed to be done here?

File: salt/modules/yumpkg.py [hunk 5]
```diff
@@ -924,6 +925,35 @@ def list_upgrades(refresh=True, **kwargs):
 list_updates = salt.utils.alias_function(list_upgrades, 'list_updates')
 
 
+def list_downloaded():
+    '''
+    .. versionadded:: Nitrogen
```
Comment (mirceaulinic) [inline salt/modules/yumpkg.py]: As this is still not merged, I would actually say to set this as `.. versionadded:: 2017.7.0` (the rest of the modules in this branch have been renamed in this PR: https://github.com/saltstack/salt/pull/41746). I should have suggested this from the first time, sorry.

Comment (isbm) [inline salt/modules/yumpkg.py]: @mirceaulinic it is all your fault. :wink: Done!

**Pull request merged**